internal/commands/cnab.go

```diff
@@ -161,3 +179,37 @@ func resolveBundle(dockerCli command.Cli, name string, pullRef bool, insecureReg
 	}
 	return nil, fmt.Errorf("could not resolve bundle %q", name)
 }
+
+func requiresBindMount(targetContextName string, targetOrchestrator string, dockerCli command.Cli) (bool, error) {
+	if targetOrchestrator == "kubernetes" {
+		return false, nil
+	}
+
+	// TODO:smarter handling of default context required
+	if targetContextName == "" {
+		return true, nil
+	}
+
+	ctxMeta, err := dockerCli.ContextStore().GetContextMetadata(targetContextName)
+	if err != nil {
+		return false, err
+	}
+	dockerCtx, err := command.GetDockerContext(ctxMeta)
+	if err != nil {
+		return false, err
+	}
+	if dockerCtx.StackOrchestrator == command.OrchestratorKubernetes {
+		return false, nil
+	}
+	dockerEndpoint, err := docker.EndpointFromContext(ctxMeta)
+	if err != nil {
+		return false, err
+	}
+
+	return isDockerHostLocal(dockerEndpoint), nil
+}
+
+func isDockerHostLocal(dockerEndpoint context.EndpointMetaBase) bool {
+	host := dockerEndpoint.Host
+	return host == "" || host == "unix:///var/run/docker.sock" || host == "npipe:////./pipe/docker_engine"
```

Comment:
I think you should test for prefixes of `unix://` or `npipe://` rather than the full string since I could e.g. run `dockerd` with `-H unix://my/socket` (if I were feeling obtuse).